module/Activity/src/Activity/Service/ActivityQuery.php

```diff
@@ -222,6 +221,7 @@ public function getUpcomingActivities($category = null, $user = null)
 
         $activityMapper = $this->getServiceManager()->get('activity_mapper_activity');
         if ($category === 'my') {
```

Comment:
I think it is better to add some explicit acl checks somewhere such that `getIdentity()` is not called for non-users. The behaviour of `getIdentity()` isn't exactly obvious in that case, so more explicit checks can prevent future mistakes.